inc/Localization/Component.php

```diff
@@ -30,7 +61,10 @@ public function get_slug() : string {
 	 * Adds the action and filter hooks to integrate with WordPress.
 	 */
 	public function initialize() {
-		add_action( 'after_setup_theme', array( $this, 'action_load_textdomain' ), 1 );
+		// Only add textdomain actions if there are .pot files.
+		if ( $this->pot_files_exist() ) {
```

Comment:
@ataylorme Just double-checking, is it really okay to run this logic on every page load?

Reply:
@felixarntz if we don't do the check then `action_load_textdomain` could run without any valid files. Is that acceptable? If not, can you provide an alternative solution?

Reply:
Might be a dumb idea, but would caching help here?

Reply:
I think storing the result of `pot_files_exist` in a cache, like a transient, would be less efficient than just running the function

Reply:
WordPress Core has logic that checks if the file is readable/exists and, if not, doesn't load so we don't need this logic.